### PR TITLE
conversion of complex_root_of to algebraic

### PR DESCRIPTION
**Pull request by videlec**
### :books: Description

As mentioned in [this sage-devel thread](https://groups.google.com/g/sage-support/c/Fybdtxbui0Y) the symbolic expression `complex_root_of(...)` do not convert to `QQbar` (with a very misleading error)
```python
sage: QQbar(complex_root_of(x^2 + 1, 0))
Traceback (most recent call last):
...
ValueError: too many values to unpack (expected 1)
```
We implement the proper hook in `sage.symbolic.expression_conversions.AlgebraicConverter` so that it now works.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/symbolic/expression_conversions.py [hunk 1]
```diff
@@ -1245,6 +1245,16 @@ def composition(self, ex, operator):
...
             0.4338837391175581? + 0.?e-18*I
 
+            sage: x = SR.var('x')
+            sage: a.composition(complex_root_of(x^3 - x^2 - x - 1, 0), complex_root_of)
+            1.839286755214161?
+            sage: a.composition(complex_root_of(x^5 - 1, 3), complex_root_of)
+            0.3090169943749474? - 0.9510565162951536?*I
+            sage: a.composition(complex_root_of(x^2 + 1, 0), complex_root_of)
+            1.?e-684 - 0.9999999999999999?*I
+            sage: a.composition(complex_root_of(x^2 + 1, 1), complex_root_of)
```
Comment (fchapoton) [inline src/sage/symbolic/expression_conversions.py:1255]: why this duplicated doctest ?

Comment (videlec) [reply src/sage/symbolic/expression_conversions.py:1255]: These are two different lines. In the first one index is `0` (corresponding to `-I`) and the second one is `1` (corresponding to `+I`).

Comment (fchapoton) [reply src/sage/symbolic/expression_conversions.py:1255]: oops, désolé

**Comment by fchapoton** [comment]
Ca m'a l'air correct. Tu as d'autres choses à changer, ou je passe en positif ?

**Comment by videlec** [comment]
> Ca m'a l'air correct. Tu as d'autres choses à changer, ou je passe en positif ?

C'est bon pour moi!

**Comment by fchapoton** [comment]
ok, allons-y

**Review by fchapoton — APPROVED** [review]
ok

**Pull request merged**